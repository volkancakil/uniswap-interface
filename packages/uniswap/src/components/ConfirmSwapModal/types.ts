export enum StepStatus {
  Preview,
  Active,
  InProgress,
  Complete,
}
