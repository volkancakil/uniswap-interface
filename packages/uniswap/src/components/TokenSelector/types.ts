import { ReactNode } from 'react'
import { TradeableAsset } from 'uniswap/src/entities/assets'
import { UniverseChainId } from 'uniswap/src/features/chains/types'
import { CurrencyInfo } from 'uniswap/src/features/dataApi/types'
import { FiatNumberType } from 'utilities/src/format/types'

export type TokenOption = {
  currencyInfo: CurrencyInfo
  quantity: number | null // float representation of balance, returned by data-api
  balanceUSD: Maybe<number>
  isUnsupported?: boolean
}

export type OnSelectCurrency = (currency: CurrencyInfo, section: TokenSection, index: number) => void

export enum TokenOptionSection {
  YourTokens = 'yourTokens',
  PopularTokens = 'popularTokens',
  RecentTokens = 'recentTokens',
  FavoriteTokens = 'favoriteTokens',
  SearchResults = 'searchResults',
  SuggestedTokens = 'suggestedTokens',
  BridgingTokens = 'bridgingTokens',
}

export type TokenSection = {
  data: TokenOption[] | TokenOption[][]
  sectionKey: TokenOptionSection
  name?: string
  rightElement?: JSX.Element
  endElement?: JSX.Element
}

export type TokenSectionsHookProps = {
  activeAccountAddress?: string
  chainFilter: UniverseChainId | null
  input?: TradeableAsset
  isKeyboardOpen?: boolean
}

export type ConvertFiatAmountFormattedCallback = (
  fromAmount: Maybe<string | number>,
  numberType: FiatNumberType,
  placeholder?: string | undefined,
) => string

export enum TokenSelectorFlow {
  Swap,
  Send,
}

export interface TokenItemWrapperProps {
  children: ReactNode
  tokenInfo: {
    address: string
    chain: number
    isNative: boolean
  }
}
