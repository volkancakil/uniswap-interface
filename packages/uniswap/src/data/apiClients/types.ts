import { useQuery } from '@tanstack/react-query'

export type UseQueryApiHelperHookArgs<REQ, RES> = {
  params?: REQ
} & Omit<Parameters<typeof useQuery<RES>>[0], 'queryKey' | 'queryFn'>

export type UseQueryWithImmediateGarbageCollectionApiHelperHookArgs<REQ, RES> = Omit<
  UseQueryApiHelperHookArgs<REQ, RES>,
  'gcTime'
> & { immediateGcTime?: number }
