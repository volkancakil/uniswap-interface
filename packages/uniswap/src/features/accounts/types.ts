export enum AccountType {
  SignerMnemonic = 'signerMnemonic', // Key lives in native keystore
  Readonly = 'readonly', // Accounts without keys (e.g. so user can track balances)
}

export interface SignerMnemonicAccountMeta {
  type: AccountType.SignerMnemonic
  address: Address
}

export interface ReadOnlyAccountMeta {
  type: AccountType.Readonly
  address: Address
}

export type AccountMeta = SignerMnemonicAccountMeta | ReadOnlyAccountMeta
