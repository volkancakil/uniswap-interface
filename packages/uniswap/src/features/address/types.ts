import { AccountType } from 'uniswap/src/features/accounts/types'

export interface SearchableRecipient {
  address: Address
  name?: string | null
  isUnitag?: boolean
  type?: AccountType
}
