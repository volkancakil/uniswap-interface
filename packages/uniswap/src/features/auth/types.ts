export type AuthTrigger = (args: { successCallback: () => void; failureCallback: () => void }) => Promise<void>
