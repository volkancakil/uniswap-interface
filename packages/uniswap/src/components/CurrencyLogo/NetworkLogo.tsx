import React from 'react'
// eslint-disable-next-line no-restricted-imports
import type { ImageSourcePropType } from 'react-native'
import { Flex, FlexProps, Image, useSporeColors } from 'ui/src'
import { ALL_NETWORKS_LOGO, ALL_NETWORKS_LOGO_UNICHAIN } from 'ui/src/assets'
import { iconSizes, zIndexes } from 'ui/src/theme'
import { getChainInfo } from 'uniswap/src/features/chains/chainInfo'
import { UniverseChainId } from 'uniswap/src/features/chains/types'
import { FeatureFlags } from 'uniswap/src/features/gating/flags'
import { useFeatureFlag } from 'uniswap/src/features/gating/hooks'
import { isMobileWeb } from 'utilities/src/platform'

export const SQUIRCLE_BORDER_RADIUS_RATIO = 0.3

type NetworkLogoProps = FlexProps & {
  chainId: UniverseChainId | null // null signifies this is the AllNetworks logo
  size?: number
  shape?: 'circle' | 'square'
  borderWidth?: number
  borderRadius?: number
}

export function TransactionSummaryNetworkLogo({
  chainId,
  size = iconSizes.icon20,
}: Pick<NetworkLogoProps, 'chainId' | 'size'>): JSX.Element {
  return <NetworkLogo borderWidth={1.5} chainId={chainId} shape="square" size={size} />
}

function _NetworkLogo({
  chainId,
  shape,
  size: sizeWithoutBorder = iconSizes.icon20,
  borderWidth = 0,
  borderRadius,
}: NetworkLogoProps): JSX.Element | null {
  const size = sizeWithoutBorder + 2 * borderWidth
  const shapeBorderRadius = shape === 'circle' ? size / 2 : size * SQUIRCLE_BORDER_RADIUS_RATIO
  const colors = useSporeColors()
  const unichainPromoEnabled = useFeatureFlag(FeatureFlags.UnichainPromo)

  const imageStyle = {
    width: size,
    height: size,
    borderRadius: borderRadius ?? shapeBorderRadius,
    borderWidth,
    borderColor: colors.surface1.val,
  }

  if (chainId === null) {
    const logo = unichainPromoEnabled ? ALL_NETWORKS_LOGO_UNICHAIN : ALL_NETWORKS_LOGO

    return (
      <Flex testID="all-networks-logo">
        <NetworkImage logo={logo} imageSize={size} />
      </Flex>
    )
  }

  const logo = getChainInfo(chainId).logo
  const imageSize = size - borderWidth * 2 // this prevents the border from cutting off the logo

  return logo ? (
    <Flex testID="network-logo" overflow="hidden" style={imageStyle} zIndex={zIndexes.mask}>
      <NetworkImage logo={logo} imageSize={imageSize} />
    </Flex>
  ) : null
}

function NetworkImage({ logo, imageSize }: { logo: ImageSourcePropType; imageSize: number }): JSX.Element {
  // As of iOS 18.3 network logos are no longer displaying because react-native-web-lite
  // adds z-index: -1 to the image. This is a workaround to display the logos on mobile web.
  return isMobileWeb && typeof logo === 'string' ? (
    <img src={logo} style={{ width: imageSize, height: imageSize }} />
  ) : (
    <Image resizeMode="contain" source={logo} width={imageSize} height={imageSize} />
  )
}

export const NetworkLogo = React.memo(_NetworkLogo)
