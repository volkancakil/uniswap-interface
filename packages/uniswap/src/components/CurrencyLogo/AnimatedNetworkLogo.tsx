import { useMemo } from 'react'
import { AnimateInOrder } from 'ui/src/animations/components/AnimateInOrder'
import { Flex } from 'ui/src/components/layout/Flex'
import { NetworkLogo } from 'uniswap/src/components/CurrencyLogo/NetworkLogo'
import { NewTag } from 'uniswap/src/components/pill/NewTag'
import { useEnabledChains } from 'uniswap/src/features/chains/hooks/useEnabledChains'
import { UniverseChainId } from 'uniswap/src/features/chains/types'

const ANIMATION_DELAY = 1500

export function AnimatedNetworkLogo({
  promoChainId,
  size,
  selectedChain,
  includeAllNetworks,
}: {
  promoChainId: UniverseChainId
  size: number
  selectedChain: UniverseChainId | null
  includeAllNetworks: boolean | undefined
}): JSX.Element {
  const { defaultChainId } = useEnabledChains()

  const selectedChainLogo = useMemo(() => {
    return <NetworkLogo chainId={selectedChain ?? (includeAllNetworks ? null : defaultChainId)} size={size} />
  }, [defaultChainId, includeAllNetworks, selectedChain, size])

  return (
    <Flex>
      <Flex animation="125ms" enterStyle={{ opacity: 0 }}>
        {selectedChainLogo}
      </Flex>
      <AnimateInOrder index={2} delayMs={ANIMATION_DELAY} position="absolute" zIndex={1}>
        <Flex animation="125ms" enterStyle={{ opacity: 0 }}>
          <NetworkLogo chainId={promoChainId} size={size} />
        </Flex>
      </AnimateInOrder>
      <AnimateInOrder index={3} delayMs={ANIMATION_DELAY} position="absolute" zIndex={1}>
        <Flex
          row
          animation="125ms"
          enterStyle={{ opacity: 0 }}
          position="relative"
          justifyContent="center"
          backgroundColor="$surface2"
        >
          <AnimateInOrder index={1} delayMs={ANIMATION_DELAY} animation="125ms" enterStyle={{ width: 0 }}>
            <Flex position="absolute" right={4} bottom={0}>
              <NewTag />
            </Flex>
          </AnimateInOrder>
          {selectedChainLogo}
        </Flex>
      </AnimateInOrder>
    </Flex>
  )
}
