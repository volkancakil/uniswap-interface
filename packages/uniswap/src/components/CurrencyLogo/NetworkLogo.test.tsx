import { NetworkLogo, TransactionSummaryNetworkLogo } from 'uniswap/src/components/CurrencyLogo/NetworkLogo'
import { UniverseChainId } from 'uniswap/src/features/chains/types'
import { render } from 'uniswap/src/test/test-utils'

jest.mock('uniswap/src/features/chains/chainInfo', () => {
  const actualChains = jest.requireActual('uniswap/src/features/chains/chainInfo')

  return {
    ...actualChains,
    getChainInfo: (chainId: unknown): unknown => {
      if (chainId === 'chainWithoutLogo') {
        return {
          logo: undefined, // no logo - for testing
        }
      } else {
        return actualChains.UNIVERSE_CHAIN_INFO[chainId as UniverseChainId]
      }
    },
  }
})

describe('NetworkLogo', () => {
  const ACTUAL_CHAIN_INFO = jest.requireActual('uniswap/src/features/chains/chainInfo').UNIVERSE_CHAIN_INFO

  it('renders without error', () => {
    const tree = render(<NetworkLogo chainId={UniverseChainId.Base} />)

    expect(tree).toMatchSnapshot()
  })

  it('renders logo when the chain info has a logo', () => {
    Object.keys(ACTUAL_CHAIN_INFO).forEach((chainId) => {
      const tree = render(<NetworkLogo chainId={chainId as unknown as UniverseChainId} />)
      expect(tree.toJSON()).toBeTruthy()
    })
  })

  it('renders null when chain info has no logo', () => {
    const tree = render(<NetworkLogo chainId={'chainWithoutLogo' as unknown as UniverseChainId} />)

    expect(tree.toJSON()).toBeNull()
  })
})

describe(TransactionSummaryNetworkLogo, () => {
  it('renders without error', () => {
    const tree = render(<TransactionSummaryNetworkLogo chainId={UniverseChainId.Base} />)

    expect(tree).toMatchSnapshot()
  })
})
