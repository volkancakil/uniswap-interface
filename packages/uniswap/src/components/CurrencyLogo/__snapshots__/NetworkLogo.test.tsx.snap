// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkLogo renders without error 1`] = `
<View
  style={
    {
      "borderColor": "#131313",
      "borderRadius": 6,
      "borderStyle": "solid",
      "borderWidth": 0,
      "flexDirection": "column",
      "height": 20,
      "overflow": "hidden",
      "width": 20,
      "zIndex": 10,
    }
  }
  testID="network-logo"
>
  <Image
    resizeMode="contain"
    source="mock-asset-BASE_LOGO.png"
    style={
      {
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;

exports[`TransactionSummaryNetworkLogo renders without error 1`] = `
<View
  style={
    {
      "borderColor": "#131313",
      "borderRadius": 6.8999999999999995,
      "borderStyle": "solid",
      "borderWidth": 1.5,
      "flexDirection": "column",
      "height": 23,
      "overflow": "hidden",
      "width": 23,
      "zIndex": 10,
    }
  }
  testID="network-logo"
>
  <Image
    resizeMode="contain"
    source="mock-asset-BASE_LOGO.png"
    style={
      {
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;
